### PR TITLE
Add support for exec subcommand

### PR DESCRIPTION
**Pull request by gaborbernat**
Run arbitrary commands within your tox environment. Don't abuse it.

Resolves https://github.com/tox-dev/tox/issues/1790

### CODE REVIEW

File: docs/changelog/1790.feature.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Add ``exec`` subcommand that allows users to run arbitrary command within the tox environment (without needing to modify
```
Comment (jugmac00) [inline docs/changelog/1790.feature.rst:1]:
```suggestion
Add ``exec`` subcommand that allows users to run an arbitrary command within the tox environment (without needing to modify
```

Comment (jugmac00) [reply docs/changelog/1790.feature.rst:1]: Is it supposed to run exactly one command?

I tried some variations of quoting to run two commands, but failed :-)

**Review by jugmac00 — COMMENTED** [review]
While there are several potentially dangerous configuration options (e.g. delete directories) already present in tox, this one is special - let's hope for the best :+1: 

@asottile Any opinion on this one?

I am not super happy with the autogenerated documentation, as at least I could not see that the command needs to be passed in after a double dash `--`.
Screenshot from 2021-08-07 11-19-43

Is there anything we could do with the autogenerated docs or maybe should we add a FAQ entry? Although - technically it is not a frequently asked question :-)

File: docs/changelog/1790.feature.rst (hunk 1 above)
Comment (gaborbernat) [reply docs/changelog/1790.feature.rst:1]: Yeah, single run only. Hence in the description ``execute an arbitrary command``.

**Comment by gaborbernat** [comment]
> I am not super happy with the autogenerated documentation, as at least I could not see that the command needs to be passed in after a double dash `--`.

It's documented in the epilogue of the sub-command (see tox4 `e --help`), if it's not rendering means we need to fix that in our documentation generation.

**Comment by jugmac00** [comment]
> > I am not super happy with the autogenerated documentation, as at least I could not see that the command needs to be passed in after a double dash `--`.
> 
> It's documented in the epilogue of the sub-command (see tox4 `e --help`), if it's not rendering means we need to fix that in our documentation generation.

Thanks - `tox4 exec --help` indeed renders the example - the documentation does not - yet.

**Comment by gaborbernat** [comment]
Created https://github.com/gaborbernat/sphinx-argparse-cli/issues/20 to address that.

**Comment by gaborbernat** [comment]
> While there are several potentially dangerous configuration options (e.g. delete directories) already present in tox, this one is special - let's hope for the best

I don't think it's overly dangerous. Users can already invoke commands within the tox environments (as in `.tox/py39/bin/pip install x`). The only thing this adds is convenience to easier invocation for that, and allowing you to take advantage of tox environment isolation and altering. And we're all consenting adults here at the end of the day, not?

**Pull request merged**